### DISCUSSION TITLE
Ark LSP in .qmd documents in positron

### DISCUSSION THREAD

**Discussion by MichelNivard** [quarto-dev/quarto-cli#10194] [Q&A] [answered]
### Description

It seems line in .qmd documents in positron the autocomplete features (which I think are part of Ark?) aren't available in r chunks in .qmd documents? They are in notebooks and .r documents obviously.

**Comment by mcanouil** [comment]
Sorry. I don't understand what you are saying or asking here or the exact context.

**Comment by cderv** [comment]
.qmd support is Positron is brought up by [Quarto VSCODE extension](https://marketplace.visualstudio.com/items?itemName=quarto.quarto) that lives in https://github.com/quarto-dev/quarto

Do you have it installed ? it does not come installed by default in Positron.

**Comment by MichelNivard** [comment]
I am saying there seems to be some kind of clash between the Quarto VSCODE extensions (i.e. the help pane etc) , and the LSP build into positron such that when I am in quarto chunks and in visual model the code auto complete suggestions provided by positron don't always work, and tab results in what I perceive as unintended behaviour. The entire line is indented when I tab which the cursor between the quotes in read.csv2(file=""), which I'd expect to suggest files in the corrent folder (what the behaviour in source mode and in .r  documents)

![image](https://github.com/quarto-dev/quarto-cli/assets/11858442/8bfdd485-318c-41b4-bcfd-eccd804486d5)

![image](https://github.com/quarto-dev/quarto-cli/assets/11858442/8ef34e99-0f9b-4634-81f6-25b3b08f93fe)

![image](https://github.com/quarto-dev/quarto-cli/assets/11858442/2be1b2f5-82c7-4818-9f55-05d6db9e9d2f)

**Reply by mcanouil** [reply]
Thank you for the clarification of the issue and the screenshots.
Very helpful!

**Reply by MichelNivard** [reply]
Further input: in RStudio, when editing a .qmd in visual mode hitting the tab does work as I's expect (it suggest contextual info, when i am between quotes it'll suggest local file paths)

**Comment by MichelNivard** [comment]
To me the tab behaviour in source and visual mode seems inconsistent, but I guess to an extend (certainly outside the code blocks) the tab having word processor like behaviour makes a lot of sense?

**Comment by juliasilge** [comment, marked as answer]
I believe you may be running into a couple of these issues:

- https://github.com/posit-dev/positron/issues/1572
- https://github.com/posit-dev/positron/issues/1805

If you have additional context to add to those issues @MichelNivard we would welcome that!

**Comment by MichelNivard** [comment]
Hi @juliasilge i think you are right, i had searched but didn’t find a prior issue. I’ll close this issue and see if I have any specific experience not covered in the other two issues.
